### PR TITLE
Broken link in primitive types list - #10445

### PR DESCRIPTION
**Pull request by jeff303**
Fix link to text type

## Summary of the changes / Why this improves CrateDB

Broken links fixed

## Checklist

 - [N/A] Added an entry in `CHANGES.txt` for user facing changes
 - [N/A] Updated documentation & `sql_features` table for user facing changes
 - [N/A] Touched code is covered by tests
 - [Y] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [Y] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by mfussenegger** [comment]
Thank you for the PR. Looks good :+1: 


> I tried installing sphinx via pip locally, and running that same command from the source root (sphinx-build -n -W -c docs/ -b html -E docs/ docs/out/html), but got ModuleNotFoundError: No module named 'crate'. So I suspect the bootstrap.sh file is doing something, but I can't find what that is. Happy to update CONTRIBUTING.rst too if I can figure out how to do this locally.

The `bootstrap.py` file additionally installs the dependencies from `requirements.txt`.

Did you see the `write_docs.rst`? 

https://github.com/crate/crate/blob/7996106dd2b6c9757346794af5ed6e228ed62b73/devs/docs/write_docs.rst#L17-L36

**Comment by mfussenegger** [comment]
ok to test

**Review by mfussenegger — APPROVED** [review]

**Comment by mfussenegger** [comment]
Looks like there is a problem with the change. Sphinx fails:

```
docs/general/ddl/data-types.rst:7:Duplicate explicit target name: "text".
```

File: docs/general/ddl/data-types.rst [hunk 1]
```diff
@@ -40,7 +40,7 @@ or collections.
 * `bigint <numeric types_>`_
 * `real <numeric types_>`_
 * `double precision <numeric types_>`_
-* `text`_
+* `text  <_data_type_text>`_
```
Comment (mechanomi) [inline docs/general/ddl/data-types.rst:43]:
```suggestion
* :ref:`text  <data-type-text>`
```

File: docs/general/ddl/data-types.rst [hunk 2]
```diff
@@ -159,7 +159,7 @@ will be truncated to ``n`` characters without raising an error.
     SELECT 1 row in set (... sec)
 
 ``character varying`` and ``varchar`` without the length specifier are
-aliases for the `text <data-type-text>`_ data type. See ` type aliases
+aliases for the `text <_data-type-text>`_ data type. See ` type aliases
```
Comment (mechanomi) [inline docs/general/ddl/data-types.rst:162]:
```suggestion
aliases for the :ref:`text <data-type-text>` data type. See ` type aliases
```

**Review by mechanomi — COMMENTED** [review]
converting these to ref links should fix the error (btw the leading underscore is a part of the label syntax, but is not included in the label identifier as is used)

**Comment by jeff303** [comment]
> Did you see the `write_docs.rst`?
> 
> https://github.com/crate/crate/blob/7996106dd2b6c9757346794af5ed6e228ed62b73/devs/docs/write_docs.rst#L17-L36

Thanks, somehow I missed this even though I saw the references in the Jenkinsfile.  I got this error when running `bootstrap.sh`:

```ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

crate-docs-theme 0.5.79 requires Sphinx==1.7.4, but you'll have sphinx 1.8.5 which is incompatible.```

I tried adding the ` --use-feature=2020-resolver` arg to the last line of `bootstrap.sh` (after `pip install`), but that didn't change anything.  Upon running `bin/sphinx dev`, I get this

```+--------- manually triggered build ---------------------------------------------
| Running Sphinx v1.8.5

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/Users/jeff/dev/crate/blackbox/.venv/lib/python3.6/site-packages/sphinx/config.py", line 368, in eval_config_file
    execfile_(filename, namespace)
  File "/Users/jeff/dev/crate/blackbox/.venv/lib/python3.6/site-packages/sphinx/util/pycompat.py", line 150, in execfile_
    exec_(code, _globals)
  File "/Users/jeff/dev/crate/docs/conf.py", line 1, in <module>
    from crate.theme.rtd.conf.crate_reference import *
ModuleNotFoundError: No module named 'crate.theme.rtd.conf.crate_reference'```

and a 404 error when opening the port number it mentions.

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**